File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let count=0;
    let length=numbers.length;
    
    dfs(0,-1);
    function dfs(num, index){
        if(index===length-1){
            if(num===target) count++;
            return;
        }
        else{
            dfs(num+numbers[index+1], index+1);
            dfs(num-numbers[index+1], index+1);
        }
    }
    return count;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    List<Integer> list = new ArrayList<>();
    public int solution(int[] numbers, int target) {
        
        dfs(0, 0, numbers);
        
        int answer=0;
        for(int n : list){
            if(n==target) answer++;
        }
        return answer;
    }
    
     public void dfs(int num, int idx, int[] numbers){
        if(idx>=numbers.length){
            list.add(num);
            return;
        }

        dfs(num+numbers[idx], idx+1, numbers);
        dfs(num-numbers[idx], idx+1, numbers);
    }
}